### PR TITLE
test: [M3-6549] - Add Cypress Integration tests for OAuth Apps

### PR DESCRIPTION
**Pull request by cliu-akamai**
## Description 📝
Add new cypress tests for OAuth Apps

## Major Changes 🔄
- Add `OAuth Apps` addition tests.
- Add `OAuth Apps` editing  tests.
- Add `OAuth Apps` resetting tests.
- Add `OAuth Apps` deletion tests.

## How to test 🧪
`yarn cy:run -s "cypress/e2e/profile/*.spec.ts"`

### CODE REVIEW

File: packages/manager/cypress/support/util/random.ts [hunk 1]
```diff
@@ -226,3 +226,23 @@ export const randomUuid = (): string => {
...
+ * Returns a random secret of fixed length.
+ *
+ * @param length - Length of the secret strings.
+ *
+ * @returns Random secret.
+ */
+export const randomSecret = (length: number = 64): string => {
+  const hexNumber = '0123456789abcdef';
+
+  const characterSelection = [hexNumber].join('').split('');
```
Comment (jdamore-linode) [inline packages/manager/cypress/support/util/random.ts:240]:
```suggestion
  const characterSelection = hexNumber.split('');
```
I think this could be simplified a bit! Is there anything major that distinguishes this from the `randomString()` util? If I'm understanding correctly we might be able to use `randomString(64)`

**Review by jdamore-linode — CHANGES_REQUESTED** [review]
Great start Cassie! I haven't been able to actually run your tests right now due to unrelated issues so I will circle back later this afternoon once everything's working, but in the meantime I left a few suggestions

File: packages/api-v4/src/account/types.ts [hunk 2]
```diff
@@ -373,6 +373,7 @@ export interface OAuthClient {
   thumbnail_url: string | null;
   public: boolean;
   status: 'disabled' | 'active' | 'suspended';
+  secret: string | null;
```
[the remark this answers is not in the document]
Comment (cliu-akamai) [reply packages/api-v4/src/account/types.ts:376]: Got it.

**Review by jaalah-akamai — APPROVED** [review]

File: packages/manager/cypress/e2e/profile/add-oauth-app.spec.ts
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/profile/add-oauth-app.spec.ts:1]: This test doesn't seem to be using mock data and is actually creating real OAuth apps. I think we're going to want some mock utils in `intercepts/profile.ts` for this since we don't want Cypress to actually create and delete OAuth apps (mostly due to concerns with having the secrets exposed in the recordings).

**Comment by hkhalil-akamai** [comment]
Looks really good. Just one minor concern: the tests do not check that the dialog/drawer for the add/edit/reset/delete actions actually close when the action is completed/cancelled (e.g., by checking that the dialog/drawer is no longer rendered or visible).

Also, should the tests try closing the dialog/drawer using the 'X' in the corner?
Screenshot 2023-06-08 at 6 54 00 PM
Screenshot 2023-06-08 at 6 54 32 PM

**Comment by cliu-akamai** [comment]
> Looks really good. Just one minor concern: the tests do not check that the dialog/drawer for the add/edit/reset/delete actions actually close when the action is completed/cancelled (e.g., by checking that the dialog/drawer is no longer rendered or visible).
> 
> Also, should the tests try closing the dialog/drawer using the 'X' in the corner? Screenshot 2023-06-08 at 6 54 00 PM Screenshot 2023-06-08 at 6 54 32 PM

That's a good point, I will update that.

File: packages/manager/cypress/e2e/profile/add-oauth-app.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    .should('be.enabled')
+    .click();
+  ui.drawer
+    .findByTitle('Create OAuth App')
+    .should('be.visible')
+    .within(() => {
+      fbltClick('Label').clear().type(oauthApp.label);
+      fbltClick('Callback URL').clear().type(oauthApp.redirect_uri);
+    });
+  ui.drawerCloseButton.find().click();
```
Comment (hkhalil-akamai) [inline packages/manager/cypress/e2e/profile/add-oauth-app.spec.ts:59]: Thank you for testing the 'X' button in Add and Edit, can you add this check to the Delete and Reset dialog as well?

**Review by hkhalil-akamai — APPROVED** [review]
Thanks for making the changes! I think you might have missed testing the 'X' button in Delete and Reset. It would be great if you could add those checks, but I will approve now so you don't have to wait on me!

File: packages/manager/cypress/support/api/profile.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * Deletes all oauth apps which are prefixed with the test entity prefix.
+ *
+ * @returns Promise that resolves when oauth apps have been deleted.
+ */
+export const deleteAllTestOAuthApps = async (): Promise<void> => {
+  const oauthApps = await depaginate<OAuthClient>((page: number) =>
+    getOAuthClients({ page_size: pageSize, page })
+  );
+
+  console.log(`oauth apps: ${JSON.stringify(oauthApps)}`);
```
Comment (jdamore-linode) [inline packages/manager/cypress/support/api/profile.ts:20]:
```suggestion
```
I don't think this is too big a deal since the secret gets redacted but probably good to get rid of this just in case!

File: packages/manager/cypress/support/ui/common.ts [hunk 5]
```diff
@@ -87,6 +88,7 @@ export const deleteAllTestData = async () => {
       deleteAllTestAccessKeys(),
       deleteAllTestTags(),
       deleteAllTestVolumes(),
+      deleteAllTestOAuthApps(),
```
Comment (jdamore-linode) [inline packages/manager/cypress/support/ui/common.ts:91]: I don't think we really need this since the tests don't make OAuth apps anymore, but it doesn't hurt anything by being there either

**Review by jdamore-linode — APPROVED** [review]
Nice work @cliu-akamai !
